### PR TITLE
update client config secret with auto build node selector

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -39,3 +39,4 @@ osbs_build_json_dir: /usr/share/osbs
 osbs_sources_command: fedpkg sources
 osbs_vendor: Fedora Project
 osbs_nodeselector: ''
+osbs_auto_build_node_selector: 'auto_build=true'
```
Comment (lcarva) [inline defaults/main.yml]: Let's have an empty value as default since that's the "safer" value.

File: templates/client-config-secret.conf.j2 [hunk 2]
```diff
@@ -57,3 +57,5 @@ enable_v1 = true
 
 [platform:aarch64]
 architecture = arm64
+
+auto_build_node_selector = {{ osbs_auto_build_node_selector }}
```
Comment (lcarva) [inline templates/client-config-secret.conf.j2]: This should be conditional, only set it if value is truthy.
More importantly, this is being defined under the wrong config section.

We need to move this over so it's under the `[{{ cluster.name }}]` section(s).
Thinking about it, this should actually be defined per cluster (`cluster.auto_build_node_selector`).

File: defaults/main.yml (hunk 1 above)
Comment (rcerven) [reply defaults/main.yml]: deleted

File: templates/client-config-secret.conf.j2 (hunk 2 above)
Comment (rcerven) [reply templates/client-config-secret.conf.j2]: ok made it defined per cluster

File: templates/client-config-secret.conf.j2 [hunk 3]
```diff
@@ -9,6 +9,9 @@ openshift_url = {{ cluster.openshift_url }}
 token_file = /var/run/secrets/atomic-reactor/{{ cluster.name | replace('_', '-') }}-orchestrator/token
 use_auth = {{ cluster.use_auth | default('true') }}
 verify_ssl = {{ cluster.verify_ssl | default('true') }}
+{% if cluster.auto_build_node_selector %}
```
Comment (lcarva) [inline templates/client-config-secret.conf.j2]: Won't this cause an exception if `cluster` does not have `auto_build_node_selector` attribute? Might need to use `default('')` filter here.

**Comment by lcarva** [comment]
Can we update the test playbook to verify it behaves correctly when `<cluster>.osbs_node_selector_labels` is set, and when it's not set?

File: templates/client-config-secret.conf.j2 (hunk 3 above)
Comment (rcerven) [reply templates/client-config-secret.conf.j2]: ok changed

**Comment by rcerven** [comment]
option added to all configuration and is omitted in minimal

File: tests/test.yml [hunk 4]
```diff
@@ -169,14 +169,14 @@
     command: >
       oc -n test-non-admin get role osbs-custom-build
     register: cmd_role
-    failed_when: "'No resources found' not in cmd_role.stderr"
+    failed_when: "'No resources found' in cmd_role.stderr"
```
Comment (twaugh) [inline tests/test.yml]: Not sure I understand why this is being changed.

The task is to make sure custom builds roles are not created in the non-admin namespace. The implementation is to run oc to try getting roles.

If there are none there, isn't that a success?

Comment (lcarva) [reply tests/test.yml]: You're right. This seems to be reversed. The tests were failing prior to this change which led us to think a newer version of `oc` is reporting the same state differently.

@rcerven, we need to investigate this a bit more.

Comment (rcerven) [reply tests/test.yml]: ok found out that in new version of openshift it actually has different error message : 
    "stderr": "Error from server (NotFound): role \"osbs-custom-build\" not found", 
so I'll check for that as well

File: tests/test.yml [hunk 5]
```diff
@@ -169,14 +169,14 @@
     command: >
       oc -n test-non-admin get role osbs-custom-build
     register: cmd_role
-    failed_when: "'No resources found' not in cmd_role.stderr"
+    failed_when: ('No resources found' not in cmd_role.stderr) and ('NotFound' not in cmd_role.stderr)
```
Comment (twaugh) [inline tests/test.yml]: Ideally these test.yml changes would be in their own commit, since they're not technically related to the auto build node selector change.

Comment (rcerven) [reply tests/test.yml]: made it in another commit
